core/amber/src/main/scala/edu/uci/ics/texera/web/resource/GoogleResource.java

```diff
@@ -0,0 +1,83 @@
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.List;
+
+public class GoogleResource {
+    private static final String APPLICATION_NAME = "Texera";
+    private static final String TOKENS_DIRECTORY_PATH = WebUtils.getTokenPath();
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+    /**
+     * Global instance of the scopes required by this quickstart.
+     * If modifying these scopes, delete your previously saved tokens/ folder.
+     */
+    private static final String CREDENTIALS_FILE_PATH = WebUtils.getCredentialPath();
+    private static final List<String> SCOPES = Arrays.asList(SheetsScopes.SPREADSHEETS, DriveScopes.DRIVE);
+
+    public static Sheets createSheetService() throws IOException {
+        try {
+            NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+            Credential credential = getCredentials(httpTransport);
+            return new Sheets.Builder(httpTransport, JSON_FACTORY, credential)
+                    .setApplicationName(APPLICATION_NAME)
+                    .build();
+        } catch (GeneralSecurityException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static Drive createDriveService() throws IOException {
+        try {
+            NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+            Credential credential = getCredentials(httpTransport);
+            return new Drive.Builder(httpTransport, JSON_FACTORY, credential)
+                    .setApplicationName(APPLICATION_NAME)
+                    .build();
+        } catch (GeneralSecurityException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     * Creates an authorized Credential object for the service account
+     */
+    private static Credential getCredentials(NetHttpTransport httpTransport) throws IOException {
+        // Load client secrets test
+        File initialFile = new File(CREDENTIALS_FILE_PATH);
+        InputStream targetStream = new FileInputStream(initialFile);
+        if (targetStream == null) {
+            throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+        }
+        GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(targetStream));
+
+        // Build flow and trigger user authorization request.
+        GoogleAuthorizationCodeFlow flow = new GoogleAuthorizationCodeFlow.Builder(
+                httpTransport, JSON_FACTORY, clientSecrets, SCOPES)
+                .setDataStoreFactory(new FileDataStoreFactory(new File(TOKENS_DIRECTORY_PATH)))
+                .setAccessType("offline")
+                .build();
+        LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8888).build();
+        return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
```

Comment:
why is this `user-id` hard coded to user?

Reply:
I am still confusing about this part as well. These codes were copied from the example provided by google sheet API and they didn't provide any explanation. I also checked the google Auth API and found they use different ways to get authentication. Thus I prefer to keep it same as the example.